### ISSUE TITLE
Enhance TC-DLand to use updated NHC land definition data

### ISSUE THREAD

**Issue by JohnHalleyGotway** [dtcenter/MET#2751]
## Describe the Enhancement ##
MET Issue #2550 added this land definition file `data/tc_data/v2023-04-07_gdland_table.dat`. It is an ASCII representation of distance to land that is read by the TC-Diag Python code in `scripts/python/tc_diag`. This overlaps with the functionality provided by the TC-DLand tool.

The tasks for this issue include:
- [ ] Check with @musgrave-kate in January 2024 to acquire update land definition files in use at NHC.
- [ ] Replace the existing `aland.dat`, `wland.dat`, and `shland.dat` data files in the `data/tc_data` directory with the updated file(s).
- [ ] Make any changes necessary in TC-DLand to use the updated datasets.
- [ ] Carefully compare the distance to land values between `v2023-04-07_gdland_table.dat` and the newly generate TC-DLand output and address any differences.
- [ ] Regenerate the existing `dland_global_tenth_degree.nc` and `dland_nw_hem_tenth_degree.nc` data files using the updated land data file.
- [ ] Coordinate with @robertdemariacira to enhance the TC-Diag Python code to use the TC-DLand output rather than `v2023-04-07_gdland_table.dat` and remove `v2023-04-07_gdland_table.dat` from the repository.

### Time Estimate ###
3 days?

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
None needed.

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
Charge NOAA R2O HAFS TnE (2784603)

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required:
  - Requires data from @musgrave-kate
  - Recommend asking @KathrynNewman to test

### Labels ###
- [x] Review default **alert** labels
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Milestone and Projects ###
- [x] Select **Milestone** as the next official version or **Backlog of Development Ideas**
- [x] For the next official version, select the **MET-X.Y.Z Development** project

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
Note that this will change the output of existing METplus use cases. But no changes should be needed to any other METplus repos.

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next official version
Select: **MET-X.Y.Z Development** project for development toward the next official release
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by jvigh** [comment]
@JohnHalleyGotway  @KathrynNewman: Here's the link to the new NHC look up table: 
https://rammb-data.cira.colostate.edu/~rdemaria/gdland/v2023-04-07_gdland_table.dat

**Comment by JohnHalleyGotway** [comment]
As discussed with @KathrynNewman during the HAFS R2O meeting, this work can be done with that funding source and included in the MET-12.0.0 release. The issue has been updated to charge project 2784603.

While doing this, please also update the MET documentation to link to more information about the ATCF file format:
https://ftp.nhc.noaa.gov/atcf/README

And provide a more clear definition of the ADECK, BDECK, EDECK terminology used primarily at NOAA/NHC, as requested by @mkavulich.

**Comment by JohnHalleyGotway** [comment]
Discussed during the HAFS project meeting on Sept 3, 2024. Once @KathrynNewman locates/provides the updated land data file, @JohnHalleyGotway can update TC-DLand to process it.
